### PR TITLE
Remove TinyDB as a dependency

### PR DESCRIPTION
**Pull request by BLKSerene**
### What does this changes

This PR removes `TinyDB` as a dependency.

### What was wrong

`TinyDB` is an unnecessary external dependency.

### How this fixes it

Remove `TinyDB` as a dependency and use `json` in the Python standard library instead.

### Your checklist for this pull request
🚨Please review the [guidelines for contributing](https://github.com/PyThaiNLP/pythainlp/blob/dev/CONTRIBUTING.md) to this repository.

- [X] Passed code styles and structures
- [x] Passed code linting checks and unit test

### CODE REVIEW

**Comment by BLKSerene** [comment]
PEP 8 should be happy now, please approve running CI and review the PR.

File: pythainlp/corpus/core.py [hunk 1]
```diff
@@ -48,19 +47,17 @@ def get_corpus_db_detail(name: str, version: str = None) -> dict:
     :return: details about a corpus
     :rtype: dict
     """
-    if _CHECK_MODE == "1":
-        local_db = TinyDB(corpus_db_path(), access_mode='r')
-    else:
-        local_db = TinyDB(corpus_db_path())
-    query = Query()
+    with open(corpus_db_path(), "r", encoding="utf_8") as f:
```
Comment (wannaphong) [inline pythainlp/corpus/core.py]: Can you change to `utf-8`?

File: pythainlp/corpus/core.py [hunk 2]
```diff
@@ -433,44 +432,41 @@ def download(
...
+                local_db["_default"][str(corpus_no)] = {
+                    "name": name,
+                    "version": version,
+                    "filename": file_name,
+                    "is_folder": is_folder,
+                    "foldername": foldername
+                }
+
+            with open(corpus_db_path(), "w", encoding="utf_8") as f:
+                json.dump(local_db, f)
```
Comment (wannaphong) [inline pythainlp/corpus/core.py]: I think you should add `ensure_ascii=False` because the file have thai.

File: pythainlp/corpus/core.py [hunk 3]
```diff
@@ -516,11 +514,13 @@ def remove(name: str) -> bool:
             shutil.rmtree(path, ignore_errors=True)
         else:
             os.remove(path)
-        db.remove(query.name == name)
-        db.close()
+        for i, corpus in db["_default"].copy().items():
+            if corpus["name"] == name:
+                del db['_default'][i]
+        with open(corpus_db_path(), 'w', encoding='utf_8') as f:
+            json.dump(db, f)
```
Comment (wannaphong) [inline pythainlp/corpus/core.py]: I think you should add `ensure_ascii=False`.

**Review by wannaphong — CHANGES_REQUESTED** [review]
Thank you for this pr! There are my review. The CI is not pass because `json.dump` doesn't save as `utf-8`.

**Comment by BLKSerene** [comment]
Initialization of the local database file fixed. Now should be fully compatible with the behavior of `TinyDB`.
The CI has passed, except for one test relating to ConceptNet API, it seem to be a network error from the API provider, hence irrelevant to this PR. https://api.conceptnet.io/c/th/%E0%B8%A3%E0%B8%B1%E0%B8%81 returns code 500.

**Review by wannaphong — APPROVED** [review]

**Comment by wannaphong** [comment]
Awesome Work!

Thank you.

**Pull request merged**